crates/network-scanner-net/src/runtime.rs

```diff
@@ -106,14 +106,18 @@ impl Socket2Runtime {
                 let mut events_happened = HashMap::new();
 
                 loop {
-                    if is_terminated.load(Ordering::Acquire) {
+                    let Some(runtime) = runtime.upgrade() else {
```

Comment:
I suspect this will work, as upgrates still [increases the Arc count ](https://github.com/rust-lang/rust/blob/11f32b73e0dc9287e305b5b9980d24aecdc8c17f/library/alloc/src/sync.rs#L2781).

Once it turns into an Arc, the drop function will not be called, and hence the notify will never happen